contracts/Fed.sol

```diff
@@ -0,0 +1,69 @@
+pragma solidity ^0.5.16;
+
+import "./SafeMath.sol";
+import "./CErc20.sol";
+import "./ERC20.sol";
+import "./Exponential.sol";
+
+contract Fed {
+    using SafeMath for uint;
+
+    CErc20 public ctoken;
+    ERC20 public underlying;
+    address public chair; // Fed Chair
+    address public gov;
+    uint public supply;
+
+    event Expansion(uint amount);
+    event Contraction(uint amount);
+
+    constructor(CErc20 ctoken_, address gov_) public {
+        ctoken = ctoken_;
+        underlying = ERC20(ctoken_.underlying());
+        underlying.approve(address(ctoken), uint(-1));
+        chair = msg.sender;
+        gov = gov_;
+    }
+
+    function changeGov(address newGov_) public {
+        require(msg.sender == gov, "ONLY GOV");
+        gov = newGov_;
+    }
+
+    function changeChair(address newChair_) public {
+        require(msg.sender == gov, "ONLY GOV");
+        chair = newChair_;
+    }
+
+    function resign() public {
+        require(msg.sender == chair, "ONLY CHAIR");
+        chair = address(0);
+    }
+
+    function expansion(uint amount) public {
+        require(msg.sender == chair, "ONLY CHAIR");
+        underlying.mint(address(this), amount);
```

Comment:
Underlying here would be DOLA?

Reply:
underlying is dola here,         underlying = ERC20(ctoken_.underlying());

Reply:
Each synth has a Fed. Dola is only the first synth. We get the synth address from its cToken

Reply:
Are we gonna keep the name `cToken` for this?

Reply:
idk tbh. We should come up with another prefix